src/core/mac/mac_frame.cpp

```diff
@@ -172,122 +172,11 @@ uint16_t Frame::GetFrameControlField(void) const
 
 otError Frame::ValidatePsdu(void) const
 {
-    otError error = OT_ERROR_PARSE;
-    uint8_t offset = kFcfSize + kDsnSize;
-    uint8_t footerLength = kFcsSize;
-    uint16_t fcf;
-
-    VerifyOrExit((offset + footerLength) <= GetPsduLength());
```

Comment:
`FindPayloadIndex()` does not include this length check, which I think is important to include.

Reply:
Added this check to` FindPayloadIndex()`, thanks.